packages/@aws-cdk/aws-lambda/lib/event-invoke-config.ts

```diff
@@ -0,0 +1,96 @@
+import { Construct, Duration, Resource } from '@aws-cdk/core';
+import { IDestination } from './destination';
+import { IFunction } from './function-base';
+import { CfnEventInvokeConfig } from './lambda.generated';
+
+/**
+ * Options to add an EventInvokeConfig to a function.
+ */
+export interface EventInvokeConfigOptions {
+  /**
+   * The destination for failed invocations.
+   *
+   * @default - no destination
+   */
+  readonly onFailure?: IDestination;
+
+  /**
+   * The destination for successful invocations.
+   *
+   * @default - no destination
+   */
+  readonly onSuccess?: IDestination;
+
+  /**
+   * The maximum age of a request that Lambda sends to a function for
+   * processing.
+   *
+   * Minimum: 60
+   * Maximum: 21600
+   *
+   * @default Duration.hours(6)
+   */
+  readonly maximumEventAge?: Duration;
+
+  /**
+   * The maximum number of times to retry when the function returns an error.
+   *
+   * Minimum: 0
+   * Maximum: 2
+   *
+   * @default 2
+   */
+  readonly maximumRetryAttemps?: number;
```

Comment:
I would name this `retryAttempts` like in the Lambda console experience.

Again from the previous guidelines https://github.com/aws/aws-cdk/blob/561cffbf9a37f0522c132e24d160ecec21e8d44d/docs/src/writing-constructs.rst#implementation-details

> Make sure you are not renaming concepts or terminology [...] because it will be hard for people to diagnose problems.

Reply:
Yep, that works.

FYI - that document has since been deleted and some of the details there is probably out of date. However, the bit you're referring to here is still accurate.